plugins/datadog/app/models/datadog_monitor_query.rb

```diff
@@ -1,24 +1,56 @@
 # frozen_string_literal: true
+require 'datadog_monitor' # prevent random load errors in dev server
+
 class DatadogMonitorQuery < ActiveRecord::Base
+  # add new handling code to datadog_monitor.rb when adding
+  MATCH_SOURCES = {
+    "Group Permalink" => "deploy_group.permalink",
+    "Group Env Value" => "deploy_group.env_value",
+    "Environment Permalink" => "environment.permalink",
+    "Cluster Permalink" => "kubernetes_cluster.permalink"
+  }.freeze
+
   belongs_to :stage, inverse_of: :datadog_monitor_queries
```

Comment:
since this is on the stage, why not support match_source to be freetext matching, and interpolate specials like $KUBERNETES_CLUSTER_PERMALINK ?

that way if none of the permalinks match you can still use the feature.

Reply:
new PR ... don't need that for now I think